rules/basic-rules.yaml

```diff
@@ -113,3 +113,23 @@
 - name: CVE-2021-4034, possible envp manipulation with empty argv
   type: Exec
   condition: payload.argc == 0
+
+
+- name: Netcat Remote Code Execution in Container
+  type: Exec
+  condition: header.container.is_some() AND
```

Comment:
This rule didn't work for me, it failed to parse

```
[2024-02-08T16:04:39Z ERROR pulsar::pulsard::module_manager] Error in module rules-engine. Module stopped. RuleParsing { filename: "/var/lib/pulsar/rules/basic-rules.yaml", error: Error { kind: SCANNER, problem: "could not find expected ':'", problem_mark: Mark { line: 135, column: 3 }, context: "while scanning a simple key", context_mark: Mark { line: 121, column: 3 } } }
```

Which makes sense, it loaded properly when I removed redundant "OR" and used `|` for multiline string in yaml:

```diff
diff --git a/rules/basic-rules.yaml b/rules/basic-rules.yaml
index 5ddaeb5..d1f5049 100644
--- a/rules/basic-rules.yaml
+++ b/rules/basic-rules.yaml
@@ -117,19 +117,20 @@

 - name: Netcat Remote Code Execution in Container
   type: Exec
-  condition: header.container.is_some() AND
-  (
+  condition: |
+    header.container.is_some() AND
     (
-      header.image ENDS_WITH "/nc" AND (
-        payload.argv CONTAINS "-e" OR
-        payload.argv CONTAINS "-c"
+      (
+        header.image ENDS_WITH "/nc" AND (
+          payload.argv CONTAINS "-e" OR
+          payload.argv CONTAINS "-c"
+        )
+      ) OR (
+        header.image ENDS_WITH "/ncat" AND (
+          payload.argv CONTAINS "--sh-exec" OR
+          payload.argv CONTAINS "--exec" OR
+          payload.argv CONTAINS "-e" OR
+          payload.argv CONTAINS "-c" OR
+          payload.argv CONTAINS "--lua-exec")
       )
-    ) OR (
-      header.image ENDS_WITH "/ncat" AND (
-        payload.argv CONTAINS "--sh-exec" OR
-        payload.argv CONTAINS "--exec" OR
-        payload.argv CONTAINS "-e" OR
-        payload.argv CONTAINS "-c" OR
-        payload.argv CONTAINS "--lua-exec") OR
     )
-  )
```

Also, can we move it to a new file (maybe `rules/container-rules.yaml`)?

Reply:
Right!! fixed

ps: there was even another error: `header.image` was wrong, it's `payload.filename` because it's an `Exec`.